### ISSUE TITLE
make test fails: test_metadata_normalization

### ISSUE THREAD

**Issue by dvzrv** [pimutils/vdirsyncer#708]
Hi!
I'm trying to build vdirsyncer for Arch Linux to move it to [community].
However, the `make test` fails with the below message:

```
=================================== FAILURES ===================================
________________ TestMemoryStorage.test_metadata_normalization _________________
tests/storage/__init__.py:301: in test_metadata_normalization
    st.none(),
E   hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 2 good ones. This will make your tests much slower, and also will probably distort the data generation quite a lot. You should adapt your strategy to filter less. This can also be caused by a low max_leaves parameter in recursive() calls
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.filter_too_much to the suppress_health_check settings for this test.
---------------------------------- Hypothesis ----------------------------------
You can add @seed(4863082389373184927620322932099709995) to this test or run pytest with --hypothesis-seed=4863082389373184927620322932099709995 to reproduce this failure.
============== 1 failed, 370 passed, 115 skipped in 22.99 seconds ==============
make: *** [Makefile:59: test] Error 1
```
My setup is:

* vdirsyncer 0.16.3 (attempted)
* python 3.6.4

**Comment by untitaker** [comment]
Did you try DETERMINISTIC_TESTS=true as described in https://vdirsyncer.pimutils.org/en/stable/packaging.html? Perhaps that helps. Will push out a bugfix release anyway.

**Comment by dvzrv** [comment]
With `make DETERMINISTIC_TESTS=true test` it lead to the same problem:

```
=================================== FAILURES ===================================
______________ TestFilesystemStorage.test_metadata_normalization _______________
tests/storage/__init__.py:301: in test_metadata_normalization
    st.none(),
E   hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 5 good ones. This will make your tests much slower, and also will probably distort the data generation quite a lot. You should adapt your strategy to filter less. This can also be caused by a low max_leaves parameter in recursive() calls
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.filter_too_much to the suppress_health_check settings for this test.
________________ TestMemoryStorage.test_metadata_normalization _________________
tests/storage/__init__.py:301: in test_metadata_normalization
    st.none(),
E   hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 5 good ones. This will make your tests much slower, and also will probably distort the data generation quite a lot. You should adapt your strategy to filter less. This can also be caused by a low max_leaves parameter in recursive() calls
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.filter_too_much to the suppress_health_check settings for this test.
```
So in the meantime I'm patching the test out to build the package.

**Comment by untitaker** [comment]
I fixed this in 0.16.4... you still really should use the DETERMINISTIC_TESTS=true option, otherwise you might get non-deterministic test failures.

**Comment by dvzrv** [comment]
@untitaker thanks!
Will do! :)
